src/compiler/crystal/codegen/codegen.cr

```diff
@@ -285,7 +285,7 @@ module Crystal
       @main = @llvm_mod.functions.add(MAIN_NAME, main_type)
       @fun_types = { {@llvm_mod, MAIN_NAME} => main_type }
 
-      if @program.has_flag? "windows"
+      if @program.has_flag?("windows") && !@program.has_flag?("gnu")
```

Comment:
**question:** I'm wondering if perhaps these conditions could be better expressed `has_flag?("msvc")` instead?
It's probably not always clear whether it's about the operating system in general or the toolchain. But here it looks more like the latter maybe?
I'm opening this as a general discussion about other occurences of windows-related flags (windows, win32, msvc). I doubt the we've always used the correct flag variants in the past.
It's probably fine to merge this PR as-is. But it seems like a good opportunity to talk and maybe correct some of it.